extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultHealthCheck.java

```diff
@@ -40,51 +38,62 @@ public HashicorpVaultHealthCheck(HashicorpVaultClient client, Monitor monitor) {
     @Override
     public HealthCheckResult get() {
 
-        HealthResponse response;
-        HealthCheckResult result;
-        try {
-            response = client.getHealth();
-        } catch (EdcException e) {  // can be thrown by the client, e.g. on JSON parsing error, etc.
-            var exceptionMsg = format(HEALTH_CHECK_ERROR_TEMPLATE, "EdcException: " + e.getMessage(), "");
-            monitor.severe(exceptionMsg, e);
-            return HealthCheckResult.failed(exceptionMsg).forComponent(COMPONENT_NAME);
+        var healthCheckResponseResult = client.doHealthCheck();
+        if (healthCheckResponseResult.failed()) {
+            monitor.severe(HEALTH_CHECK_ERROR);
+            return HealthCheckResult
+                    .failed(healthCheckResponseResult.getFailureMessages())
+                    .forComponent(HashicorpVaultExtension.NAME);
         }
 
+        var errors = new ArrayList<String>(0);
```

Comment:
spurious: no point in initializing with an initial cap of 0.

Reply:
The list is empty in 99.9% of cases (happy path) so why would you need a bigger list from the get go if java can just resize the list if needed ?

Reply:
This will have a negligible impact on runtime resource consumption, particularly since it will be dereferenced quickly. Please keep with the coding style we currently have, it's easier to read.